lib/customisation.nix

```diff
@@ -365,6 +378,35 @@ rec {
     else
       mapAttrs mkAttrOverridable pkgs;
 
+  makeAlias =
+    let
+      currentRelease = lib.versions.majorMinor lib.version;
+    in
+    self: name:
+    {
+      type,
+      package ? null,
+      reason ? null,
+      release ? null,
+    }:
+    assert type == "alias";
+    if release == null then
+      self.${package}
+    else if release != currentRelease then
+      if package == null then
+        abort "Throw for '${name}' was added in ${release}. The alias must now be removed."
+      else
+        abort "Warning for '${name}' was added in ${release}. The alias must now be converted to a throw."
+    else if package == null || !self.config.allowAliases then
+      {
+        type = "error";
+        meta = throw "'${name}' was removed because it was ${reason}.";
```

Comment:
Why the choice of `meta` specifically here? To ensure that even `nix-env` etc. are unhappy?